face-detection-explainer.md

```diff
@@ -12,66 +12,129 @@
+
+The coordinates in the members `boundingBox` and `centerPoint` of the detected faces and landmarks are defined similarly as in the
 [`pointsOfInterest`](https://w3c.github.io/mediacapture-image/#points-of-interest)
-field in [`MediaTrackSettings`](https://w3c.github.io/mediacapture-image/#dom-mediatracksettings-pointsofinterest)
-with the exception that the points may also lie outside of the frame since a detected face could be
-partially outside of the visible image.
-A `Point2D` is interpreted to represent a pixel position in a normalized square space. The origin of
-coordinates {x,y} = {0.0, 0.0} represents the upper leftmost corner whereas the {x,y} =
-{1.0, 1.0} represents the lower rightmost corner relative to the rendered frame: the x coordinate (columns) increases rightwards and the y
-coordinate (rows) increases downwards.
+member in [`MediaTrackSettings`](https://w3c.github.io/mediacapture-image/#dom-mediatracksettings-pointsofinterest)
+with the exception that the coordinates may also lie outside of the frame since a detected face could be
+partially (or in special cases even fully) outside of the visible image.
+A coordinate is interpreted to represent a location in a normalized square space. The origin of
+coordinates (x,y) = (0.0, 0.0) represents the upper leftmost corner whereas the (x,y) =
+(1.0, 1.0) represents the lower rightmost corner relative to the rendered frame: the x-coordinate (columns) increases rightwards and the y-coordinate (rows) increases downwards.
+
+### Constraints
 
-The constraint `faceDetectionMode` is used by applications to describe the level of facial data that they need. At the lowest enabled level, `presence` will return the sequence of `DetectedFace`, but the `contour` and `landmarks` sequences will be empty. When `faceDetectionMode` is `contour`, arbitrary number of points around the faces will be returned but no landmarks. An user agent might return only four contour points corresponding to face bounding box. If a Web application needs only maximum of four contour points (bounding box), it can set `faceDetectionMode` to `bounding-box` which limits number of contour points to four, residing at the corners of a rectangle around the detected faces.
+New members are added to capabilities, constraints, and settings for Web applications to enable and control face and face landmark detection with `getUserMedia()` and `applyConstraints()` and to query capabilities of face detection with `getCapabilities()` methods. Web applications should not ask more facial metadata than what they need to limit computation. For example, if an applications is content with just a face bounding box, it should set the constraint `humanFaceLandmarkDetectionMode` to `"none"`.

```

Comment:
The sentence _"Web applications should not ask more facial metadata than what they need to limit computation"_ has a typo, but also seems to be admonishing web developers to behave a certain way, which reads a bit odd to me.

Could it be rephrased to emphasize the benefits of asking for less information?

Reply:
Rephrased to "Web applications can reduce the computational load on the user agent by only requesting the necessary amount of facial data, rather than asking for more than they need."